Modules/_zoneinfo.c

```diff
@@ -694,14 +694,21 @@ zoneinfo_fromutc(PyObject *obj_self, PyObject *dt)
         }
         else {
             PyObject *replace = PyObject_GetAttrString(tmp, "replace");
+            if (replace == NULL) {
+                Py_DECREF(tmp);
+                return NULL;
+            }
             PyObject *args = PyTuple_New(0);
+            if (args == NULL) {
+                Py_DECREF(tmp);
+                Py_DECREF(replace);
+                return NULL;
+            }
             PyObject *kwargs = PyDict_New();
-
-            Py_DECREF(tmp);
-            if (args == NULL || kwargs == NULL || replace == NULL) {
-                Py_XDECREF(args);
-                Py_XDECREF(kwargs);
-                Py_XDECREF(replace);
+            if (kwargs == NULL) {
+                Py_DECREF(tmp);
```

Comment:
This does not look right; `tmp` would leak if `kwargs != NULL`. But why is not `tmp` simply decreffed after the PyObject\_GetAttrString() call in line 696? AFAICS, it is not needed after that call. We could decref it right away and simplify the error handling.